game/plugin/skills/fishing/src/fishing.plugin.kts

```diff
@@ -0,0 +1,143 @@
+            stop()
+            return
+        }
+
+        //Check inv capacity
+        if (mob.inventory.freeSlots() == 0) {
+            mob.inventory.forceCapacityExceeded()
+            stop()
+            return
+        }
+
+        //check bait
+        val bait = findBait();
+        if (findBait() == -2) {
+            mob.sendMessage("You need " + ItemDefinition.lookup(tool.bait).name.toLowerCase() + "s to fish at this spot.")
+            stop()
+            return
+        }
+
+        //Do action
+        if (started) {
+            //Get possible fish
+            var fishList = ArrayList<Fish>()
+            for (fish in fishies) {
+                if (fish.level <= mob.skillSet.getSkill(Skill.FISHING).currentLevel) {
+                    fishList.add(fish)
+                }
+            }
+
+            //Get which fish is caught
+            var fish:Fish by Delegates.notNull()
+            if (fishList.size == 1) {
+                fish = fishList.get(0)
+            } else {
+                //Get 70/30 chance
+                if (Random().nextInt(100) > 70) {
+                    fish = fishList.get(0)
+                } else {
+                    fish = fishList.get(1)
+                }
+            }
+
+            //Perform catch
+            if (successfulCatch(mob.skillSet.getSkill(Skill.FISHING).currentLevel, fish.level)) {
+                if (bait != -1)  {
+                    mob.inventory.remove(bait)
+                }
+                mob.inventory.add(fish.id)
+                val fishName = ItemDefinition.lookup(fish.id).name.toLowerCase()
+                var name = if (fishName.endsWith("s")) "some " + fishName else "a " + fishName
+                mob.sendMessage("You catch " + name + ".")
+                mob.skillSet.addExperience(Skill.FISHING, fish.exp)
+                //Check bait for continued fishing
+                if (findBait() == -2) {
+                    mob.sendMessage("You need more " + ItemDefinition.lookup(tool.bait).name.toLowerCase() + "s to fish at this spot.")
+                    stop()
+                    return
+                }
+            }
+
+        } else {
+            startFishing()
+        }
+        mob.playAnimation(tool.animation)
+    }
+}
+
+on { NpcActionMessage::class}
+        .then {
+            if (lookupFishingNpc(it.world.npcRepository[index].id) != null) {
+                it.startAction(FishingAction(it, it.world.npcRepository[index].position, lookupFishingNpc(it.world.npcRepository[index].id)!!, option))
+                this.terminate()
+            }
+        }
+
+start {
+    //TODO: Schedule fishing spots moving
+    val DELAY = 1000 //1000 ticks because I do not know how long...?
+    it.schedule(object: ScheduledTask(DELAY, false){
+        override fun execute() {
+            for (npc in FISHING_NPCS.values) {
+                npc.moveFishingSpot()
```

Comment:
```kotlin
override fun execute() = FISHING_NPCS.values().forEach(FishNpc::moveFishingSpot)
```

Reply:
I just removed this all together until I can get the locations that they move to.